wdl-lint/src/rules/redundant_none.rs

```diff
@@ -0,0 +1,106 @@
+use std::fmt::Debug;
+use rowan::ast::AstNode;
+use wdl_ast::AstNodeExt;
+use wdl_ast::AstToken;
+use wdl_ast::Diagnostic;
+use wdl_ast::Diagnostics;
+use wdl_ast::Document;
+use wdl_ast::Span;
+use wdl_ast::SupportedVersion;
+use wdl_ast::SyntaxElement;
+use wdl_ast::VisitReason;
+use wdl_ast::Visitor;
+use wdl_ast::v1::BoundDecl;
+use crate::Rule;
+use crate::Tag;
+use crate::TagSet;
+
+const ID: &str = "RedundantNoneAssignment";
+
+fn redundant_none_assignment(span: Span, name: &str) -> Diagnostic {
+    Diagnostic::note("redundant None assignment")
+        .with_rule(ID)
+        .with_highlight(span)
+        .with_fix(format!("can be shortened to {}", name))
+}
+
+#[derive(Default, Debug, Clone, Copy)]
+pub struct RedundantNoneAssignment(Option<SupportedVersion>);
+
+impl Rule for RedundantNoneAssignment {
+    fn id(&self) -> &'static str {
+        ID
+    }
+
+    fn description(&self) -> &'static str {
+        "Flags redundant None assignments to optional inputs."
+    }
+
+    fn explanation(&self) -> &'static str {
+        "Optional inputs with explicit None assignments can be simplified. For example, \
+         String? foo = None can be shortened to String? foo."
```

Comment:
@a-frantz - I don't particularly like this snippet from the spec (the variable names, that is), but showing that the two are equivalent could be helpful as part of this text.  I've made a simple edit suggestion below, but I'm curious if you think a longer `explanation` is preferable.

```
    # the following are equivalent undefined optional declarations
    String? maybe_five_but_is_not
    String? also_maybe_five_but_is_not = None
```

```suggestion
         `String? foo = None` is equivalent to and can be shortened to `String? foo`."
```